dts/bindings/crc/st,stm32-crc.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2024 Brill Power Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+description: ST STM32 family CRC
+
+compatible: "st,stm32-crc"
+
+include: [base.yaml]
+
+properties:
+  reg:
+    required: true
+
+  clocks:
+    type: phandle-array
+    required: true
+    description: Clock information
+
+  timeout:
+    type: int
+    default: 0
+    description: |
+      Timeout period (milliseconds) for use when acquiring mutex lock for 8-bit CRC
+      in thread-safe mode.
+      Min=0, Max=65535, >10000=K_FOREVER
```

Comment:
As this is pure SW behavior, it should rather be configured by Kconfig.
https://docs.zephyrproject.org/latest/build/dts/dt-vs-kconfig.html
> In other words, devicetree mainly deals with hardware, and Kconfig with software.

A reason to use device tree could that you want to be able to configure different values per instance, but this doesn't look the case here.